### DISCUSSION TITLE
Openai已经支持按照自定义的json结构化输出了,千问后续会有这种功能吗

### DISCUSSION THREAD

**Discussion by taosu0216** [alibaba/higress#1195] [Announcements]
在ai rag的最后一步还是获取数据库的内容然后再调用对话模型输出,这里啥时候能支持自定义json结构然后在填充输出
![image](https://github.com/user-attachments/assets/9df7608c-fdc2-4d3c-ba32-eab9dfc36eeb)

**Comment by johnlanni** [comment]
qwen对这个的支持可以给阿里云百炼API平台提个工单问一下哈。

工程上到是可以做一个插件来支持按照自定义json结构输出，但正确性很难保证，可能意义不大。

**Comment by johnlanni** [comment]
这个issue我转到discussion了

**Comment by Suchun-sv** [comment]
这个是不是能放在ai-proxy的openai的接口改进那里？感觉挺有用的。

**Reply by johnlanni** [reply]
我的想法是可以单独做一个插件，把llm结果里的json提取出来转换为application/json格式的响应，提取出来后还可以做一下字段格式校验，如果字段格式不对，可以带上错误信息再次请求llm。

**Reply by taosu0216** [reply]
这个库 https://github.com/instructor-ai/instructor-go 好像提供了go通过jsonschema的方法来让openai生成结构化的返回结果,如果能把这个库的功能添加进ai rag里面感觉会很棒

**Reply by taosu0216** [reply]
我目前测试了十几次 , 正确率百分比 , 通过这个库调接口输出的内容得到的内容都是自定义的结构化的结果 , 但是不知道怎么跟现有的ai rag融合起来
